e2e-tests/src/test/kick_out.rs

```diff
@@ -0,0 +1,158 @@
+/// appropriate reason.
+pub fn kick_out_automatic(config: &Config) -> anyhow::Result<()> {
+    let root_connection = config.create_root_connection();
+
+    let validator_keys = get_test_validators(config);
+    let reserved_validators = account_ids_from_keys(&validator_keys.reserved);
+    let non_reserved_validators = account_ids_from_keys(&validator_keys.non_reserved);
+
+    let seats = CommitteeSeats {
+        reserved_seats: 2,
+        non_reserved_seats: 2,
+    };
+
+    change_validators(
+        &root_connection,
+        Some(reserved_validators.clone()),
+        Some(non_reserved_validators.clone()),
+        Some(seats),
+        XtStatus::InBlock,
+    );
+
+    wait_for_full_era_completion(&root_connection)?;
+
+    let start_era_validators = get_current_era_validators(&root_connection);
+
+    assert_eq!(start_era_validators.reserved, reserved_validators);
+    assert_eq!(start_era_validators.non_reserved, non_reserved_validators);
+
+    let committee_kick_out_config = get_committee_kick_out_config(&root_connection);
+
+    assert_eq!(
+        committee_kick_out_config.minimal_expected_performance,
+        DEFAULT_KICK_OUT_MINIMAL_EXPECTED_PERFORMANCE
+    );
+    assert_eq!(
+        committee_kick_out_config.underperformed_session_count_threshold,
+        DEFAULT_KICK_OUT_SESSION_COUNT_THRESHOLD
+    );
+    assert_eq!(
+        committee_kick_out_config.clean_session_counter_delay,
+        DEFAULT_CLEAN_SESSION_COUNTER_DELAY
+    );
+
+    let validator_to_disable =
+        &non_reserved_validators[VALIDATOR_TO_DISABLE_NON_RESERVED_INDEX as usize];
+    info!(target: "aleph-client", "Validator to disable: {}", validator_to_disable);
+
+    let underperformed_validator_session_count =
+        get_underperformed_validator_session_count(&root_connection, validator_to_disable);
+
+    assert_eq!(underperformed_validator_session_count, 0);
+
+    let validator_to_disable_kick_out_reason =
+        get_kick_out_reason_for_validator(&root_connection, validator_to_disable);
+
+    assert!(validator_to_disable_kick_out_reason.is_none());
+
+    let validator_seed = get_validator_seed(VALIDATOR_TO_DISABLE_OVERALL_INDEX);
+    let stash_controller = NodeKeys::from(validator_seed);
+    let controller_key_to_disable = stash_controller.controller;
+
+    // This connection has to be set up with the controller key.
+    let connection_to_disable =
+        SignedConnection::new(NODE_TO_DISABLE_ADDRESS, controller_key_to_disable);
+
+    set_invalid_keys_for_validator(&connection_to_disable)?;
+
+    let current_session = get_current_session(&root_connection);
+
+    wait_for_at_least_session(
+        &root_connection,
+        current_session + SESSIONS_TO_MEET_KICK_OUT_THRESHOLD,
+    )?;
+
+    let underperformed_validator_session_count =
+        get_underperformed_validator_session_count(&root_connection, validator_to_disable);
+
+    assert_eq!(underperformed_validator_session_count, 0);
```

Comment:
Shouldn't it be 3?

Reply:
My thought initially as well, but it turns out that once the counter hits `3`, it is immediately reset to `0`. Thanks to @Marcin-Radecki for clarifying that.

Reply:
Maybe add a comment explaining that?

Reply:
Sure, added :)